packages/matter.js/src/storage/Storage.ts

```diff
@@ -12,11 +13,34 @@ export class StorageError extends MatterError {}
  * Matter.js uses this key/value API to manage persistent state.
  */
 export interface Storage {
-    initialize(): Promise<void>;
-    close(): Promise<void>;
-    get<T extends SupportedStorageTypes>(contexts: string[], key: string): T | undefined;
-    set<T extends SupportedStorageTypes>(contexts: string[], key: string, value: T): void;
-    delete(contexts: string[], key: string): void;
-    keys(contexts: string[]): string[];
-    clearAll(contexts: string[]): void;
+    initialize(): MaybePromise<void>;
+    close(): MaybePromise<void>;
+    get(contexts: string[], key: string): MaybePromise<SupportedStorageTypes | undefined>;
+    set(contexts: string[], key: string, value: SupportedStorageTypes): MaybePromise<void>;
+    delete(contexts: string[], key: string): MaybePromise<void>;
+    keys(contexts: string[]): MaybePromise<string[]>;
+    contexts(contexts: string[]): MaybePromise<string[]>;
+    clearAll(contexts: string[]): MaybePromise<void>;
+}
+
+export abstract class MaybeAsyncStorage implements Storage {
```

Comment:
Huh, could get rid of this type given that the base storage class is inherently "maybe async"?  So just have Storage and SyncStorage?  That feels like it'd be cleaner.

Reply:
Thats because of https://github.com/microsoft/TypeScript/issues/57905. The "Storage" needs to be without the generic, else anything screwes up TS wise. We can kill this once it is fixed.. I will add this as a reference